src/sinol_make/commands/run/__init__.py

```diff
@@ -315,20 +315,33 @@ def compile_solutions(self, solutions):
         os.makedirs(self.COMPILATION_DIR, exist_ok=True)
         os.makedirs(self.EXECUTABLES_DIR, exist_ok=True)
         print("Compiling %d solutions..." % len(solutions))
+        args = [(solution, True) for solution in solutions]
         with mp.Pool(self.cpus) as pool:
-            compilation_results = pool.map(self.compile, solutions)
+            compilation_results = pool.starmap(self.compile, args)
         return compilation_results
 
 
-    def compile(self, solution):
+    def compile(self, solution, use_extras = False):
         compile_log_file = os.path.join(
             self.COMPILATION_DIR, "%s.compile_log" % package_util.get_file_name(solution))
         source_file = os.path.join(os.getcwd(), "prog", self.get_solution_from_exe(solution))
         output = os.path.join(self.EXECUTABLES_DIR, package_util.get_executable(solution))
 
+        extra_compilation_args = []
+        extra_compilation_files = []
+        if use_extras:
+            if "extra_compilation_args" in self.config:
+                lang = os.path.splitext(source_file)[1][1:]
+                for file in self.config["extra_compilation_args"].get(lang, []):
+                    extra_compilation_args.append(os.path.join(os.getcwd(), "prog", file))
+
+            for file in self.config.get("extra_compilation_files", []):
+                extra_compilation_files.append(os.path.join(os.getcwd(), "prog", file))
+
         try:
             compile.compile(source_file, output, self.compilers,
-                            open(compile_log_file, "w"), self.args.weak_compilation_flags)
+                            open(compile_log_file, "w"), self.args.weak_compilation_flags, extra_compilation_args,
```

Comment:
By the way, shouldn't that be a `with open`? It seems that this file descriptor is not being closed, right?

Reply:
Yes, it should